pkg/generate/app/sourcelookup.go

```diff
@@ -44,45 +44,32 @@ type SourceRepository struct {
 // NewSourceRepository creates a reference to a local or remote source code repository from
 // a URL or path.
 func NewSourceRepository(s string) (*SourceRepository, error) {
-	var location *url.URL
-	switch {
-	case strings.HasPrefix(s, "git@"):
-		base := "git://" + strings.TrimPrefix(s, "git@")
-		url, err := url.Parse(base)
-		if err != nil {
-			return nil, err
-		}
-		location = url
+	uri, err := url.Parse(s)
```

Comment:
Previous code didn't work either, because if you pass eg. `git@github.com:openshift/origin` it'll change it to `git://github.com:openshift/origin` which isn't valid repo URL (note the colon after .com). With my changes it's working properly. Other option would be to change `SourceRepository` structure not to contain `URL` yet still use `url.Parse` for validating it.